docs/source/gs-processing/usage/emr.rst

```diff
@@ -77,177 +77,93 @@ inline policy:
     INSTANCE_TYPE=m6i.4xlarge
     # INSTANCE_TYPE=m6g.4xlarge # Use for arm64 image
     REGION=us-east-1
-    EMR_VERSION="emr-7.0.0"
-    CORE_INSTANCE_COUNT=3
+    CORE_INSTANCE_COUNT=2
     CLUSTER_NAME="${USER}-gsp-${CORE_INSTANCE_COUNT}x-${INSTANCE_TYPE}"
-    INSTANCE_ROLE="EMR_EC2_DefaultRole"
-    TERMINATION_HOURS=1
-
-    # We assume this script is saved in the same path as container-executor.json
-    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-
-    LOG_BUCKET=my-log-bucket
-
-    aws emr create-cluster \
-        --applications Name=Hadoop Name=Spark \
-        --auto-termination-policy IdleTimeout=$((${TERMINATION_HOURS}*60*60)) \
-        --configurations file://${SCRIPT_DIR}/container-executor.json \
-        --ec2-attributes KeyName=${KEYPAIR},SubnetId=${SUBNET_ID},AdditionalMasterSecurityGroups=${MASTER_SG} \
-        --instance-groups InstanceGroupType=MASTER,InstanceCount=1,InstanceType=${INSTANCE_TYPE} \
-            InstanceGroupType=CORE,InstanceCount=${CORE_INSTANCE_COUNT},InstanceType=${INSTANCE_TYPE} \
-        --log-uri s3://${LOG_BUCKET}/emr-logs/ \
-        --name ${CLUSTER_NAME} \
-        --region ${REGION} \
-        --release-label ${EMR_VERSION} \
-        --use-default-roles
-
-Running the above will return a JSON structure like:
-
-.. code-block:: json
-
-    {
-        "ClusterId": "j-XXXXXXXXXX",
-        "ClusterArn": "arn:aws:elasticmapreduce:us-east-1:<ACCOUNT>:cluster/j-XXXXXXXXXX"
-    }
-
-Make note of the cluster ID, which we will use to log into the leader instance.
-
-We can also run a waiter to ensure we only proceed to the next step when the cluster is
-ready to run jobs:
-
-.. code-block:: bash
-
-    aws emr wait cluster-running --cluster-id j-XXXXXXXXXX --region ${REGION} && echo "Cluster ready"
-
-Log in to the leader and submit a GSProcessing job
---------------------------------------------------
-
-To submit a job we can use a helper ``bash`` script, which we list below:
-
-.. code-block:: bash
-
-    # submit-gsp-job.sh
-    #!/usr/bin/env bash
-    set -euox pipefail
 
+    # GSProcessing arguments
     MY_BUCKET="enter-your-bucket-name-here"
     REGION="bucket-region" # e.g. us-west-2
     INPUT_PREFIX="s3://${MY_BUCKET}/gsprocessing-input"
-    NUM_EXECUTORS=2
     OUTPUT_BUCKET=${MY_BUCKET}
     GRAPH_NAME="small-graph"
     CONFIG_FILE="gconstruct-config.json"
+    DO_REPARTITION="true"
+    GENERATE_REVERSE="true"
 
-    ACCOUNT=$(aws sts get-caller-identity --query Account --output text)
-
-    REPOSITORY="graphstorm-processing-emr"
-    ARCH="x86_64"
-    TAG="latest-${ARCH}"
-    IMAGE="${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/${REPOSITORY}:${TAG}"
 
+    # We assume this script is saved in the same path as submit_gsp_emr_step.py
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+    # Upload the entry point to S3
+    PATH_TO_ENTRYPOINT="$SCRIPT_DIR/../graphstorm_processing/distributed_executor.py"
```

Comment:
Are you assuming we are under `graphstorm-processing/scripts` ?
Can you be more specific in the comments?

Reply:
Yes, in line 139 we mention `# We assume this script is saved in the same path as submit_gsp_emr_step.py`. I can point this out more explicitly.